lib/orbit/ocs/repo/entities.ex

```diff
@@ -0,0 +1,267 @@
+defmodule Orbit.Ocs.Entities do
+  @moduledoc """
+  Responsible for processing parsed OCS messages, and writing appropriate changes
+  to the Trains and Trips in the database.
+  Also provides querying of relevant entities for the current service date.
+  """
+  require Logger
+
+  alias Orbit.Ocs.Train
+  alias Orbit.Ocs.Trip
+  alias Orbit.RailLine
+  alias Orbit.Repo
+
+  alias Orbit.Ocs.Message
+
+  alias Orbit.Ocs.Message.{
+    TschAsnMessage,
+    TschConMessage,
+    TschDelMessage,
+    TschDstMessage,
+    TschLnkMessage,
+    TschNewMessage,
+    TschOffMessage,
+    TschRldMessage,
+    TschTagMessage
+  }
+
+  import Ecto.Query
+
+  @type entities() :: [Trip.t()]
+
+  @spec query_latest(DateTime.t()) :: entities()
+  def query_latest(current_datetime \\ DateTime.utc_now()) do
+    # TODO: What filtering/sorting makes sense for this query?
```

Comment:
question: are you thinking about filtering/sorting from a bandwidth reduction perspective or something else?

Reply:
Think it makes more sense to address trip matcher ordering in a subsequent PR.